src/main/java/org/alfresco/repo/search/impl/solr/SolrAdminClient.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2016 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software. 
+ * If the software was purchased under a paid Alfresco license, the terms of 
+ * the paid license agreement will prevail.  Otherwise, the software is 
+ * provided under the following open source license terms:
+ * 
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.search.impl.solr;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.alfresco.repo.search.impl.lucene.JSONAPIResult;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.InitializingBean;
+
+/**
+ * Solr Admin client interface
+ * Provides access to Actions and Commands in SOLR API
+ * @author aborroy
+ * @since 6.2
+ */
+public interface SolrAdminClient extends BeanFactoryAware, InitializingBean
```

Comment:
I'm confused by this - I couldn't see any classes implementing this interface.

Reply:
Finally changed the name to **SolrAdminClientInterface** to mark a difference with the implementation **SOLRAdminClient**